README.md

```diff
@@ -75,64 +81,160 @@ which the standard toast aims to accomplish natively.
 
+- The first child node,
+  typically either a text node or a container like `<p>`,
+  provides a message.
+- If more than one child node is present,
+  then the last child element,
+  which can be either an `<a>` or a `<button>`,
+  provides a call to action.
 
-### Properties & Functions
+Thus, the following would all work well out of the box:
 
-The std-toast library will come with a `showToast` function,
-which will handle the case of opening a toast
-and optionally creating it as well in Javascript.
+```html
+<std-toast>Hello world!</std-toast>
 
-#### `showToast(message, options)`
+<std-toast><p>Hello world!</p></std-toast>
 
-Creates a toast object displaying the `message`
-with an object of the `options`.
+<std-toast>
+  <p>Hello world!</p>
+  <button>Click me!</button>
+</std-toast>
 
-The options the developers can configure are:
+<std-toast>
+  <p>Hello world!</p>
+  <a href="https://example.com/" target="_blank">Click me!</button>
+</std-toast>
+```
 
-- `duration`
-- `closeButton`
-- `multiple` (??)
-- `newestOnTop` (??)
+These can equivalently be created via JavaScript:
+
+```js
+// TODO: write this once we figure out the JS API for supplying HTML messages,
+// and for supplying customizable actions
+```
+
+More complex toasts,
+that don't fit the above content model,
+would require more custom handling on the part of the developer:
+
+```html
+<std-toast>
+  <p>Hello world!</p>
+  <p>A second paragraph?!</p>
+
+  <button>Action 1</button>
+  <button>Action 2</button>
+</std-toast>
+```
 
-### Pseudo classes
+Such an unusual toast would still integrate with other toasts in terms of stacking and positioning behavior,
+and some of the default styles that are inherited may be useful.
+But the page author will need to handle the action clicks themselves (instead of using the `"actionclick"` event),
+and will need to add additional styling to handle the extra contents.
 
-- `:hover`
-- `:focus`
+TODO: when we have a prototype, link to/show an example of this in action.
 
-### Events
+#### Methods
 
-The `<std-switch>` will emit events when it is opened and when it is closed.
-*TODO(me): should it emit an event when the action is taken?*
-*TODO(me): event for onFocus?*
+The `.show(options)` function shows a toast element,
+by toggling its `open=""` attribute to true.
+The `options` include:
+
+- `duration`: how long to show the toast, in milliseconds. Defaults to ???
+- `multiple`: ???
+- `newestOnTop`: ???
+
+TODO: how do `multiple` and `newestOnTop` work?
+Should those be a global setting?
+Per container?
+How do we deal with different toasts having different values?
+
+#### Events
+
+A `<std-toast>` element can fire the following events:
+
+- `"open"`: the toast was shown
+- `"close"`: the toast was closed, either explicitly by the user, or via the timeout.
+  (Note: if animations were applied, the toast may not be entirely invisible at the time this event fires)
```

Comment:
Some toast libraries used "willClose" and "didClose" (as well as the same for open) to solve this visibility problem. I think it's worth looking into that again to see how popular that pattern was.

Reply:
Sounds good. I will add a TODO for now. My worry is if there are custom animations we'd get thrown off.